bazel-mode.el

```diff
@@ -21,6 +21,39 @@
 ;;
 ;;; Code:
 
+(defgroup bazel-mode nil
+  "Major mode for Bazel BUILD files."
+  :link '(url-link "https://github.com/bazelbuild/emacs-bazel-mode")
+  :group 'languages)
+
+(defcustom bazel-mode-buildifier-cmd "buildifier"
+  "Filename of buildifier executable."
+  :type 'file
+  :group 'bazel-mode
+  :link "https://github.com/bazelbuild/buildtools/tree/master/buildifier")
+
+(defun bazel-mode-buildifier ()
+  "Format current buffer using buildifier."
+  (interactive "*")
+  (let ((input-buffer (current-buffer))
+        (buildifier-buffer (get-buffer-create "*buildifier*"))
+        (temp-file (make-nearby-temp-file "buildifier")))
+    (with-temp-file temp-file (insert-buffer-substring input-buffer))
```

Comment:
Whereas make-temp-file allows an argument (text) which initializes the file with the contents given in this argument, with-nearby-temp-file does not. This means that I must explicitly copy the contents of the input-buffer into the temp file. write-file displays the file it writes in a buffer and marks this buffer unread (undesirable), and write-region displays the text "Wrote /tmp/buildifierabcde..." in the minibuffer (also undesirable). By using with-temp-file I can write the BUILD file contents to the temp file silently. But maybe there's a more idiomatic way to do this?

Reply:
Only `make-nearby-temp-file` provides the correct semantics for all remote files. You can use `write-region` to write arbitrary text to the file afterwards. See its documentation for how to suppress the message.

Reply:
Done.